docs/okeps/okep-5088-evpn.md

```diff
@@ -0,0 +1,789 @@
+configured EVPN fabric. Therefore VTEP IP provisioning may be done by the provider and configured for each node.
+
+The IP address assigned or found by OVN-Kubernetes will be annotated to the node as "k8s.ovn.org/<vtep name>: <ip>".
+If all nodes do not have an IP address annotated for this VTEP, the VTEP CR will go into error state.
+
+### CUDN CRD changes for EVPN
+
+This API change depends on the [No Overlay Feature](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5289) which will
+introduce a new Transport field that we can extend to use EVPN. The API change will look like this:
+
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: l3-primary
+  labels:
+    bgp: enabled
+spec:
+  namespaceSelector:
+    matchLabels:
+      kubernetes.io/metadata.name: udn-test
+  network:
+    topology: Layer2
+    layer2:
+      role: Primary
+      subnets:
+      - 10.20.100.0/16
+  transport: EVPN
+  evpnConfiguration:
+    vtep: evpn-vtep
+    macVRF:
+      vni: 100
+    ipVRF:
+      vni: 101
+```
+
+In the above example, a new transport type "EVPN" is introduced. This is paired with evpnConfiguration section, which
+specifies the name of the VTEP CR to use. We will only support specifying a single VTEP to use for EVPN.
+
+The macVRF field must be provided for Layer2 networks, and cannot be provided for Layer3. ipVRF must be provided for
+Layer3, and is optional for Layer2 if an IP-VRF is also desired for the Layer2 UDN.
+
+The "vni" specified under each will be used to determine the VNID for each EVPN segment.
+The VNI values may not overlap between any EVPN CR.
+
+Additionally, with Layer 2 MAC-VRFs stretched outside the cluster, we need a way to be able to tell OVN-Kubernetes not
+to by default allocate certain IPs from the shared subnet. The Layer2 UDN already provides "ReservedSubnets" for this
+purpose. If a VM or other application is migrated from the external fabric over the L2 VPN, then it is able to use
+a static IP from this ReservedSubnets field. However, when a VM migrates it not only needs to preserve its IP address,
+but also its default gateway. OVN-Kubernetes uses the first IP in the subnet as the default gateway for pods. This may
+not align with the gateway IP used by a VM off-cluster, and in addition we use the second IP in the subnet as the
+ovn-k8s-mpx IP that may also be used on the Layer 2 network off cluster. To address there are other fields already
+provided in the Layer2 UDN spec that can be leveraged:
+ - DefaultGatewayIPs
+ - InfrastructureSubnets
+
+These configurable fields should be considered when creating a MAC-VRF that will be extended to outside the cluster.
+
+### Route Advertisement (RA) CRD changes
+
+Note, in order to use EVPN, a Route Advertisement CR must also be created which selects the CUDN. Although no API changes
+are required for the RA, it may be extended in the future to allow for configuration of RouteTargets.
```

Comment:
could you elaborate on why we're not doing this now ? 

i.e. how is this optional ?